src/Session.php

```diff
@@ -1762,12 +1773,23 @@ public static function stopImpersonating()
 
        //store user which was impersonated by another user
         $impersonate_user = $_SESSION['glpiname'];
+        $impersonator_info = $_SESSION['impersonator_info'] ?? [];
 
         $auth = new Auth();
         $auth->auth_succeded = true;
         $auth->user = $user;
         Session::init($auth);
 
+        // Restore previous user values
+        if (!empty($impersonator_info)) {
```

Comment:
What about cleaning session info?

Reply:
`$_SESSION` variable is already cleaned in `Session::init()`.

Reply:
oh, indeed